pyo3-ffi/src/object.rs

```diff
@@ -22,19 +26,38 @@ pub const _Py_IMMORTAL_REFCNT: Py_ssize_t = {
     }
 };
 
-pub const PyObject_HEAD_INIT: PyObject = PyObject {
-    #[cfg(py_sys_config = "Py_TRACE_REFS")]
-    _ob_next: std::ptr::null_mut(),
-    #[cfg(py_sys_config = "Py_TRACE_REFS")]
-    _ob_prev: std::ptr::null_mut(),
-    #[cfg(Py_3_12)]
-    ob_refcnt: PyObjectObRefcnt { ob_refcnt: 1 },
-    #[cfg(not(Py_3_12))]
-    ob_refcnt: 1,
-    #[cfg(PyPy)]
-    ob_pypy_link: 0,
-    ob_type: std::ptr::null_mut(),
-};
+#[cfg(Py_GIL_DISABLED)]
+pub const _Py_IMMORTAL_REFCNT_LOCAL: u32 = u32::MAX;
+#[cfg(Py_GIL_DISABLED)]
+pub const _Py_REF_SHARED_SHIFT: isize = 2;
+
+pub const fn PyObject_HEAD_INIT() -> PyObject {
+    PyObject {
+        #[cfg(py_sys_config = "Py_TRACE_REFS")]
+        _ob_next: std::ptr::null_mut(),
+        #[cfg(py_sys_config = "Py_TRACE_REFS")]
+        _ob_prev: std::ptr::null_mut(),
+        #[cfg(Py_GIL_DISABLED)]
+        ob_tid: 0,
+        #[cfg(Py_GIL_DISABLED)]
+        _padding: 0,
+        #[cfg(Py_GIL_DISABLED)]
+        ob_mutex: unsafe { mem::zeroed::<PyMutex>() },
```

Comment:
I think this can be `PyMutex::new()` if we define it as above.

That avoids the `unsafe` block. And maybe allows `PyObject_HEAD_INIT` to remain a constant? Are there other fields with interior mutability?